### PR TITLE
Dev - Integrate CaImAn - Work in progress

### PR DESCRIPTION
**Pull request by kabilar**

### CODE REVIEW

File: djimaging/processing.py [hunk 1]
```diff
@@ -267,6 +267,54 @@ def make(self, key):
...
+            elif caiman_loader.params['motion']['pw_rigid'][...]:
+                nonrigid_mc = {'block_height': caiman_loader.params['motion']['strides'][...][0]+caiman_loader.params['motion']['overlaps'][...][0],
+                                'block_width': caiman_loader.params['motion']['strides'][...][1]+caiman_loader.params['motion']['overlaps'][...][1],
+                                'block_count_y': ,# number of blocks tiled - round down and cut off last stride
+                                'block_count_x': ,# number of blocks tiled
+                                'outlier_frames': }# mask with true for frames with outlier shifts (already corrected)
+                                # block vs patch naming
+                b_id=0
+                nr_blocks=[]
+                for j in range(len(caiman_loader.y_shifts_els[:,0])):
```
Comment (dimitri-yatsenko) [inline djimaging/processing.py:300]: i and j are not necessary.  Use y_shifts_els and x_shift_els as iterators.

Comment (dimitri-yatsenko) [reply djimaging/processing.py:300]: This loop can be more expressively and efficiently implemented as a list comprehension.

File: img_loaders/caiman.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    def __init__(self, caiman_filepath_automated, caiman_filepath_manual):
+        # ---- Verify files exists ----
+        if not caiman_filepath_automated.exists() or not caiman_filepath_manual.exists():
+            raise FileNotFoundError('Both "*.hdf5" files are not found. Invalid CaImAn file: {}, {}'.format(caiman_filepath_automated,caiman_filepath_manual))
+        else:
+            # ---- Open CaImAn output files ----
+            h5f_automated = h5py.File(caiman_filepath_automated, 'r')
+            h5f_manual = h5py.File(caiman_filepath_manual, 'r')
+
+            if '/mc/shifts_rig' not in h5f_manual and ...
```
Comment (dimitri-yatsenko) [inline img_loaders/caiman.py:30]:
```python
if not any(s in h5fmanual for s in ('/mc/shifts_irg', 'mc_...', ...)):
```

**Comment by ttngu207** [comment]
merging to `dev` to continue development

**Pull request merged**